### PR TITLE
Start design improvements and minor improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Prepare the app layout for upcoming features, make the design a bit more material-like, and add an option to select the theme.

Please note that this PR temporarily breaks the dark theme. This will be fixed in a follow-up PR.

### CODE REVIEW

File: src/main/kotlin/util/FilterButton.kt [hunk 1]
```diff
@@ -22,8 +22,8 @@ fun <T> FilterButton(
     buttonWidth: Dp = 130.dp,
     dropdownWidth: Dp = 130.dp,
     onFilterChange: (T) -> Unit,
-    buttonContent: @Composable RowScope.(T) -> Unit,
-    dropdownItem: @Composable RowScope.(T) -> Unit
+    buttonContent: @Composable RowScope.(selectedItem: T) -> Unit,
```
Comment (sschuberth) [inline src/main/kotlin/util/FilterButton.kt:25]: You've renamed `selectedItem` to `item` above, so shouldn't you also use `item` here?

File: src/main/kotlin/ui/dependencies/Dependencies.kt [hunk 2]
```diff
@@ -77,6 +75,7 @@ import org.ossreviewtoolkit.workbench.util.CaptionedText
 import org.ossreviewtoolkit.workbench.util.ErrorCard
 import org.ossreviewtoolkit.workbench.util.Expandable
 import org.ossreviewtoolkit.workbench.util.ExpandableText
+import org.ossreviewtoolkit.workbench.util.FilterTextField
```
Comment (sschuberth) [inline src/main/kotlin/ui/dependencies/Dependencies.kt:79]: Commit message: Can you briefly mention what's the benefit of `FilterTextField` / of reusing it?

File: src/main/kotlin/ui/Menu.kt [hunk 3]
```diff
@@ -53,7 +53,7 @@ enum class MenuItem(val icon: MaterialIcon) {
 
 @Composable
 fun Menu(currentScreen: MenuItem, apiState: OrtApiState, onSwitchScreen: (MenuItem) -> Unit) {
-    Surface(modifier = Modifier.fillMaxHeight().width(200.dp).zIndex(1F), elevation = 8.dp) {
+    Surface(modifier = Modifier.fillMaxHeight().width(200.dp).zIndex(zIndex = 3f), elevation = 8.dp) {
```
Comment (sschuberth) [inline src/main/kotlin/ui/Menu.kt:56]: Nit: Unrelated change.

File: src/main/kotlin/util/FilterButton.kt (hunk 1 above)
Comment (mnonnenmacher) [reply src/main/kotlin/util/FilterButton.kt:25]: No, the `buttonContent` is only called for the selected item, but the `dropdownItem` is called for each individual `item`.

**Review by sschuberth — APPROVED** [review]
